include/uapi/sound/sof/tokens.h

```diff
@@ -191,4 +192,9 @@
 /* COPIER */
 #define SOF_TKN_INTEL_COPIER_NODE_TYPE		1980
 
+/* pin_binding */
+#define SOF_TKN_PIN_BINDING_BIND_TYPE		2000
+#define SOF_TKN_PIN_BINDING_QUEUE_ID		2001
+#define SOF_TKN_PIN_BINDING_WNAME		2002
```

Comment:
@aiChaoSONG why do we need type, num_pin_bindings and queue ID tokens?
WHat we want really want is a set ot SOURCE_PIN_NAME's and a set of SINK_PIN_NAME's. The number of source/sink pins should equal the number of source/sink pins per widget in topology. We cannot used a hybrid methodology for assigning queue IDs. If a widget needs special binding, it should be for all source/sink pins or for none

Reply:
Another issue I have is that I cannot make this topology compiled, https://github.com/thesofproject/linux/issues/3684#issuecomment-1163865130
I tried to make several changes, but I still get compile error. 
can you help to take a look? so that I can make corresponding kernel change and debug kernel with it.
I suspect there is an issue in alsatplg.

Reply:
The branch work with current Kernel PR: https://github.com/aiChaoSONG/sof/tree/src_sink_pins
The branch cannot compile: https://github.com/thesofproject/sof/commits/src_sink_pins_v1, this version is based on your patch [0001-topology2-add-pin-info-to-copier-widget.txt](https://github.com/thesofproject/linux/files/8963235/0001-topology2-add-pin-info-to-copier-widget.txt), and of course, the patch cannot compile either.

Reply:
@ranj063  Please take a look at this document. It explains why I use these tokens, and how they helps to simplify topology. 

[on_the_queue_support_design.pdf](https://github.com/thesofproject/linux/files/9157321/on_the_queue_support_design.pdf)